src/identity-provider.js

```diff
@@ -19,18 +20,20 @@ class IdentityProvider {
     const selfSigningFn = async (id, data) => await this._keystore.sign(key, data)
     // If signing function was not passed, use keystore as the identity signer
     identitySignerFn = isDefined(identitySignerFn) ? identitySignerFn : selfSigningFn
+    // If no type was indicated, set to default
+    type = isDefined(type) ? type : defaultType
     // Sign the id with the signing key we're going to use
-    const pkSignature = await this._keystore.sign(key, id)
+    const idSignature = await this._keystore.sign(key, id)
     // Get the hex string of the public key
     const publicKey = key.getPublic('hex')
     // Sign both the key and the signature created with that key
-    const signature = await identitySignerFn(id, publicKey + pkSignature)
-    return new Identity(id, publicKey, pkSignature, signature, this)
+    const pubKeyIdSignature = await identitySignerFn(id, publicKey + idSignature)
+    return new Identity(id, publicKey, idSignature, pubKeyIdSignature, type, this)
   }
 
-  static async createIdentity (keystore, id, identitySignerFn) {
+  static async createIdentity (keystore, id, identitySignerFn, type) {
```

Comment:
Switch the order here too if you change https://github.com/orbitdb/orbit-db-identity-provider/pull/10/files#r214800880.